# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/linux/amd-i2c-smb.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- linux-6.0/include/linux/pci_ids.h.vanilla	2022-10-19 15:34:48.143490752 +0200
+++ linux-6.0/include/linux/pci_ids.h	2022-10-19 15:36:18.614482714 +0200
@@ -612,6 +612,7 @@
 #define PCI_DEVICE_ID_AMD_HUDSON2_SMBUS		0x780b
 #define PCI_DEVICE_ID_AMD_HUDSON2_IDE		0x780c
 #define PCI_DEVICE_ID_AMD_KERNCZ_SMBUS  0x790b
+#define PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS 0x43f6
 
 #define PCI_VENDOR_ID_TRIDENT		0x1023
 #define PCI_DEVICE_ID_TRIDENT_4DWAVE_DX	0x2000
--- linux-6.0/drivers/i2c/busses/i2c-piix4.c.vanilla	2022-10-19 15:36:24.688482175 +0200
+++ linux-6.0/drivers/i2c/busses/i2c-piix4.c	2022-10-19 15:51:58.268399237 +0200
@@ -408,6 +408,8 @@
 	    (PIIX4_dev->vendor == PCI_VENDOR_ID_AMD &&
 	     PIIX4_dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS &&
 	     PIIX4_dev->revision >= 0x49) ||
+            (PIIX4_dev->vendor == PCI_VENDOR_ID_AMD &&
+             PIIX4_dev->device == PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS) ||
 	    (PIIX4_dev->vendor == PCI_VENDOR_ID_HYGON &&
 	     PIIX4_dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS))
 		smb_en = 0x00;
@@ -460,6 +462,7 @@
 	if (PIIX4_dev->vendor == PCI_VENDOR_ID_AMD ||
 	    PIIX4_dev->vendor == PCI_VENDOR_ID_HYGON) {
 		if (PIIX4_dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS ||
+		    PIIX4_dev->device == PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS ||
 		    (PIIX4_dev->device == PCI_DEVICE_ID_AMD_HUDSON2_SMBUS &&
 		     PIIX4_dev->revision >= 0x1F)) {
 			piix4_port_sel_sb800 = SB800_PIIX4_PORT_IDX_KERNCZ;
@@ -907,6 +910,7 @@
 	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_SBX00_SMBUS) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_HUDSON2_SMBUS) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_KERNCZ_SMBUS) },
+	{ PCI_DEVICE(PCI_VENDOR_ID_AMD, PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_HYGON, PCI_DEVICE_ID_AMD_KERNCZ_SMBUS) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS,
 		     PCI_DEVICE_ID_SERVERWORKS_OSB4) },
@@ -994,6 +998,7 @@
 	int retval;
 
 	if (dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS ||
+	    dev->device == PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS ||
 	    (dev->device == PCI_DEVICE_ID_AMD_HUDSON2_SMBUS &&
 	     dev->revision >= 0x1F)) {
 		piix4_adapter_count = HUDSON2_MAIN_PORTS;
@@ -1097,6 +1102,7 @@
 
 	if (dev->vendor == PCI_VENDOR_ID_AMD &&
 	    (dev->device == PCI_DEVICE_ID_AMD_HUDSON2_SMBUS ||
+	     dev->device == PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS ||
 	     dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS)) {
 		retval = piix4_setup_sb800(dev, id, 1);
 	}
--- linux-6.0/drivers/i2c/busses/i2c-piix4.c.vanilla	2022-10-19 16:27:39.105549126 +0200
+++ linux-6.0/drivers/i2c/busses/i2c-piix4.c	2022-10-19 16:34:43.076544007 +0200
@@ -235,8 +235,9 @@
 	 * the same processors and is the recommended access method.
 	 */
 	return (PIIX4_dev->vendor == PCI_VENDOR_ID_AMD &&
-		PIIX4_dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS &&
-		PIIX4_dev->revision >= 0x51);
+		(PIIX4_dev->device == PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS ||
+		 (PIIX4_dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS &&
+		  PIIX4_dev->revision >= 0x51)));
 }
 
 static int piix4_setup(struct pci_dev *PIIX4_dev,
@@ -1050,7 +1051,8 @@
 
 		if ((dev->vendor == PCI_VENDOR_ID_AMD ||
 		     dev->vendor == PCI_VENDOR_ID_HYGON) &&
-		    dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS) {
+		    (dev->device == PCI_DEVICE_ID_AMD_KERNCZ_SMBUS ||
+		     dev->device == PCI_DEVICE_ID_AMD_PROMONTORY_SMBUS)) {
 			u8 imc;
 
 			/*
